var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -260,6 +266,8 @@ class Openblas(CMakePackage, MakefilePackage):
         msg="Visual Studio does not support OpenBLAS dynamic dispatch features",
     )
 
+    requires("+noavx512", when="%intel@2021:2022")
```

Comment:
If that works, we should also make the change in our fork (or get it for free when we pull in spack develop). Are you certain this is only an issue with intel@2021? Note that there is no `intel@2022`, run `spack info intel` and `spack info `intel-oneapi-compilers` to check that. The `intel@2022` is a false version that somehow creeped into some of our compiler site configs and they should be fixed. The classic compilers (labeled as `intel@` are all 2021.x.y all the way up to the latest oneAPI release 2024.2.0).

Reply:
I haven't looked at it any further than what's in the variant description. If it's a build-time error I can see about testing with a couple versions of 2021 (along with something more recent).

Reply:
I think it's fine to do a blanket no-avx512 for intel@2021, but @2022 doesn't really exist